interfaces/seccomp/backend.go

```diff
@@ -58,30 +58,39 @@ var (
 	releaseInfoId            = release.ReleaseInfo.ID
 	releaseInfoVersionId     = release.ReleaseInfo.VersionID
 	requiresSocketcall       = requiresSocketcallImpl
+	snapSeccompVersionInfo   = snapSeccompVersionInfoImpl
 )
 
-func seccompToBpfPath() string {
+func snapSeccompVersionInfoImpl(c Compiler) (string, error) {
+	return c.VersionInfo()
+}
+
+func snapSeccompPath() string {
 	// FIXME: use cmd.InternalToolPath here once:
 	//   https://github.com/snapcore/snapd/pull/3512
 	// is merged
-	snapSeccomp := filepath.Join(dirs.DistroLibExecDir, "snap-seccomp")
+	defaultSnapSeccomp := filepath.Join(dirs.DistroLibExecDir, "snap-seccomp")
 
 	exe, err := osReadlink("/proc/self/exe")
 	if err != nil {
 		logger.Noticef("cannot read /proc/self/exe: %v, using default snap-seccomp command", err)
-		return snapSeccomp
-	}
-	if !strings.HasPrefix(exe, dirs.SnapMountDir) {
```

Comment:
Curious if this was removed because it was deemed unnecessary or because there was a problem?

Reply:
There was a problem with finding snap-seccomp during early bootstrap. When snapd snap was mounted under `/tmp/tmp.XXXXX`, the code would return `$rootfs/usr/lib/snapd/snap-seccomp`, while it should actually return `/tmp/tmp.XXXXX/usr/lib/snapd/snap-seccomp`.